### DISCUSSION TITLE
How to convert to USDZ format?

### DISCUSSION THREAD

**Discussion by neciszhang** [donmccurdy/glTF-Transform#1219] [Q&A] [answered]
Hello, I would like to ask if gltf-transform can be converted to USDZ format. If not, how can I quickly convert gltf to USDZ?Thanks!

**Comment by donmccurdy** [comment, marked as answer]
glTF Transform does not read or write formats other than glTF. Currently the USD format does not have a specification, and in my view only the [official implementation](https://github.com/PixarAnimationStudios/OpenUSD) is reliable. When people refer to "convert to USDZ" they usually mean convert to a USD file tailored for the particular set of features and limitations currently implemented by Apple's proprietary viewers. Unfortunately that's a moving target and not (to my knowledge) well documented. Those caveats aside, these tools might be worth a try:

- https://github.com/google/usd_from_gltf
- https://threejs.org/editor/

The three.js editor uses THREE.GLTFLoader and THREE.USDZExporter, so you could incorporate those into your own web applications too.
